model/connection.js

```diff
@@ -101,6 +269,75 @@ class Connection {
         return true;
     }
 
+    /**
+     * Validate the Connection Feature.
+     *
+     * Name, id, connection and layer follow Validation's string rules.
+     * objects.
+     *
+     * @since 1.0.0
+     *
+     * @see validateSinkSourcePoints
+     *
+     * @returns {boolean}
+     */
+    validateFeature() {
+        let valid = super.validate();
+        valid = Validation.testStringValue(this.connection, 'connection', 'Connection') ? valid : false;
+        valid = Validation.testStringValue(this.layer, 'layer', 'Connection') ? valid : false;
+        valid = Validation.testDimensionValue(this.width, 'width', 'Connection') ? valid : false;
+        valid = Validation.testDimensionValue(this.depth, 'depth', 'Connection') ? valid : false;
+        valid = this.validateSinkSourcePoints() ? valid : false;
+        if (this.type !== Validation.DEFAULT_CON_TYPE) {
+            valid = false;
+            console.log('Connection: Field "type" is invalid. It must be "' + Validation.DEFAULT_CON_TYPE + '", but' +
+                    ' it is "' + this.type + '".');
+        }
+
+        return valid;
+    }
+
+    /**
+     * Validate the sinkPoint and sourcePoint objects.
+     *
+     * The line between sinkPoint and sourcePoint must be straight and cannot
+     * have the same x and y values. Both must not evaluate to falsey and be
+     * valid Coord objects.
+     *
+     * @since 1.0.0
+     *
+     * @see validateEndpoint
+     *
+     * @returns {boolean}
+     */
+    validateSinkSourcePoints() {
+        let valid = true;
+
+        if (!this.sourcePoint || !this.sourcePoint.validate()) {
+            console.log('Connection: Fields "sourcePoint" is invalid.');
+            return false;
+        }
+
+        if (!this.sinkPoint || !this.sinkPoint.validate()) {
+            console.log('Connection: Fields "sinkPoint" is invalid.');
+            return false;
+        }
+
+        // Check that this connection is a straight line
+        if ((this.sinkPoint.x !== this.sourcePoint.x) && (this.sinkPoint.y !== this.sourcePoint.y)) {
```

Comment:
Straight line doesn't mean rectilinear here. It can have slope it just can't be curved.

Reply:
The check has been removed.